ogr/ogrfielddefn.cpp

```diff
@@ -863,6 +863,67 @@ const char *OGRFieldDefn::GetFieldTypeName(OGRFieldType eType)
     return "(unknown)";
 }
 
+/************************************************************************/
+/*                          GetFieldTypeByName()                        */
+/************************************************************************/
+/**
+ * \brief Fetch field type by name.
+ * @param pszName the name of the field type.
+ * @return the field type or OFTString if there is no match with known type names.
+ * @since GDAL 3.9.4
+ */
+OGRFieldType OGRFieldDefn::GetFieldTypeByName(const char *pszName)
```

Comment:
Would be nice to have full testing of this function (and the one for subtypes) in autotest/cpp/test_ogr.cpp

Reply:
done